### ISSUE TITLE
Focus rings on Cards partially remain when focus is lost in Safari 15

### ISSUE THREAD

**Issue by LFDanLu** [adobe/react-spectrum#3078]
# 🐛 Bug Report 

1. Navigate to https://reactspectrum.blob.core.windows.net/reactspectrum/1d6bdd52c9d7a5cbfab84595281494f27e3af90b/storybook/index.html?path=/story/cardview-gallery-layout--default-gallery-no-thin-images in Safari (15+ perhaps, occurs in 15.4 and 15.3 for sure)
2. Select and deselect a card via keyboard
3. Navigate to an adjacent card via keyboard arrows
4. Note that a focus ring partially remains

<img width="633" alt="image" src="https://user-images.githubusercontent.com/9661127/165396078-6508862f-f0e4-485d-9a54-eec48f564a12.png">

Not 100% sure if this is a Safari 15 issue, but it wasn't caught in previous testing session so I'm guessing it is something recent. I can reproduce this issue on builds from commits 5 months back so I don't believe it was a code change that caused this.

Would be good to test after Safari 15.5 comes out since there is a `contain` fix that should be included in that version but otherwise we may have to figure out if we can do the focus ring styling in a different way.

## 🧢 Your Company/Team

RSP

**Comment by LFDanLu** [comment]
Note this happens even with the VirtualizerItem `contains` changes reverted to include `paint`

**Comment by snowystinger** [comment]
I can't reproduce anymore in 18
